### PR TITLE
fix: make keyboard navigation (s/w) iteration-aware (Issue https://github.com/pedropaulovc/codjiflo/issues/189)

### PR DESCRIPTION
**Pull request by pedropaulovc**
**Problem:**
When viewing a PR in iteration mode, keyboard shortcuts s (next file)
and w (previous file) navigated using GitHub API file count instead of
the iteration-aware file list. This caused navigation to jump to files
that weren't visible in the current iteration view.

**Root Cause:**
- selectNextFile/selectPreviousFile in useDiffStore used files.length
  from GitHub API
- FileList component displays files from useIterationAwareFiles()
- These two file lists can differ in iteration mode (filtered files,
  artifact-only files, different indices)

**Solution:**
- Move file navigation logic from store to useKeyboardShortcuts hook
- Hook now uses useIterationAwareFiles() - same data source as FileList
- Navigate through visible files using their originalIndex values
- Handles PR description (-1 index) correctly
- Works in both regular and iteration modes

**Changes:**
- src/features/keyboard/hooks/useKeyboardShortcuts.ts:
  - Import useIterationAwareFiles and PR_DESCRIPTION_INDEX
  - Implement selectNextFile/selectPreviousFile using iteration-aware files
  - Navigate by finding current position in visible files array
- e2e/iteration-keyboard-navigation.spec.ts:
  - Add E2E test for iteration-aware keyboard navigation
  - Tests s/w navigation through visible files only
  - Tests boundary conditions (first/last file)

Co-Authored-By: Claude Sonnet 4.5 <noreply@anthropic.com>

---

🔍 **[Review in CodjiFlo](https://codjiflo.vza.net/pedropaulovc/codjiflo/199)**

### CODE REVIEW

File: src/features/keyboard/hooks/useKeyboardShortcuts.ts [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
+    const currentPos = files.findIndex((f) => f.originalIndex === selectedFileIndex);
+
+    // If current file is found and not the last one, go to next
+    if (currentPos >= 0 && currentPos < files.length - 1) {
+      const nextFile = files[currentPos + 1];
+      if (nextFile) {
+        selectFile(nextFile.originalIndex);
+      }
+    }
+  }, [selectedFileIndex, files, selectFile]);
```
[the remark this answers is not in the document]
Comment (pedropaulovc) [reply src/features/keyboard/hooks/useKeyboardShortcuts.ts]: Same edge case as Thread 1. I'll update both `selectNextFile` and `selectPreviousFile` to navigate to the PR description when the current file is not in the iteration-aware list.

-- 🤖 [Claude Code](https://claude.ai/claude-code)

File: e2e/iteration-keyboard-navigation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ */
+
+import { test, expect } from './fixtures/console-warnings';
+import { isMockMode } from './fixtures/mode';
+
+test.describe('Iteration Keyboard Navigation (#189)', () => {
+  // TODO: These tests need proper mock setup with setupIterationArtifactMock() and buildIterationDb()
+  // Currently they reference a PR without mock data and fail with timeout
+  // Unit tests in useKeyboardShortcuts.test.ts provide coverage for the fix
+  test.skip(() => true, 'Needs iteration artifact mock setup - see useKeyboardShortcuts.test.ts for unit test coverage');
```
[the remark this answers is not in the document]
Comment (pedropaulovc) [reply e2e/iteration-keyboard-navigation.spec.ts]: This has been addressed - E2E tests are now fully implemented with proper iteration artifact mocks using `buildIterationDb()`. The skip has been removed and all 3 E2E tests are passing in both mock and prod modes.

-- 🤖 [Claude Code](https://claude.ai/claude-code)

File: e2e/iteration-keyboard-navigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    // Select iteration 1
+    const iterationSelector = page.getByTestId('iteration-selector');
+    await iterationSelector.click();
+    await page.getByText('1', { exact: true }).click();
+
+    await page.waitForSelector('[data-testid="file-list-item"]');
+
+    const fileListItems = page.getByTestId('file-list-item');
+    const visibleFileCount = await fileListItems.count();
+    console.log(`Found ${visibleFileCount} visible files in iteration 1`);
```
[the remark this answers is not in the document]
Comment (pedropaulovc) [reply e2e/iteration-keyboard-navigation.spec.ts]: Already removed during rebase - the E2E tests were completely rewritten with proper mocks and this debug statement no longer exists.

-- 🤖 [Claude Code](https://claude.ai/claude-code)
